src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -190,30 +194,31 @@ private ImmutableArray<BoundPattern> BindSubPropertyPatterns(PropertyPatternSynt
                         diagnostics.Add(node, useSiteDiagnostics);
                     }
 
-                    properties.Add(property);
-                    pattern = this.BindPattern(syntax.Pattern, null, property.GetTypeOrReturnType(), hasErrors, diagnostics);
+                    var propertyType = property.GetTypeOrReturnType();
+                    pattern = this.BindPattern(syntax.Pattern, null, propertyType, hasErrors, diagnostics);
+                    result.Add(new BoundSubPropertyPattern(syntax, propertyType, property, resultKind, pattern, hasErrors));
                 }
                 else
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMember, propName, type, propName.Identifier.ValueText);
                     pattern = new BoundWildcardPattern(node, hasErrors: true);
+                    result.Add(new BoundSubPropertyPattern(syntax, CreateErrorType(), null, resultKind, pattern, true));
                 }
-
-                boundPatternsBuilder.Add(pattern);
             }
 
-            return boundPatternsBuilder.ToImmutableAndFree();
+            return result.ToImmutableAndFree();
         }
 
-        private Symbol FindPropertyByName(TypeSymbol type, IdentifierNameSyntax name, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        private Symbol FindPropertyByName(TypeSymbol type, IdentifierNameSyntax name, out LookupResultKind resultKind, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
             var symbols = ArrayBuilder<Symbol>.GetInstance();
-            var result = LookupResult.GetInstance();
-            this.LookupMembersWithFallback(result, type, name.Identifier.ValueText, arity: 0, useSiteDiagnostics: ref useSiteDiagnostics);
+            var lookupResult = LookupResult.GetInstance();
+            this.LookupMembersWithFallback(lookupResult, type, name.Identifier.ValueText, arity: 0, useSiteDiagnostics: ref useSiteDiagnostics);
+            resultKind = lookupResult.Kind;
 
-            if (result.IsMultiViable)
+            if (lookupResult.IsMultiViable)
             {
-                foreach (var symbol in result.Symbols)
+                foreach (var symbol in lookupResult.Symbols)
```

Comment:
We can even have more than one with no parameters due to multiple interface inheritance. This code should probably defer to lookup code elsewhere to do the disambiguation, and then give an error if the result isn't as expected.